### PR TITLE
Preserve data for Book and Label classes

### LINKED ISSUE #13: [1pt] Preserve Data [TM #1]
All data should be preserved by saving collections in .json files.

### LINKED ISSUE #15: [3pt] Create a schema.sql file with tables [TM #1]
- Create a schema.sql file with tables that will be analogical to the structure of the classes that you created:
   - books table (add all properties and associations from the parent Item class as table columns)
   - labels table

### PR DESCRIPTION
**Pull request by rubydevi**
### In this PR I have made the following changes:
#13 
- All data for `Book` and `Labels` are preserved by saving collections in .json files. 

#15 
- Created tables that will be analogical to the structure of the classes created:
  - books table
  - labels table

### CODE REVIEW

**Review by Yonashenok — CHANGES_REQUESTED** [review]
# Hi @rubydevi 

You're have done a great job!!!

## Highlights
- All JSON files are properly created ✔
- Data is properly saved and retrieved ✔
- All the tabel are working fine ✔
- Correct use of Git Flow ✔
- All linter checks are passing ✔
- Descriptive PR title with a great summary ✔

## Required Changes ♻️
- kindly Check the the `File.write` for` label.json` file
- you just have to make a few changes and you're on your way 🚀.




Cheers and Happy coding!👏👏👏

File: src/modules/book_loader.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  end
+
+  def load_labels
+    File.new('data/labels.json', 'w') unless File.exist?('data/labels.json')
+    labels = File.read('data/labels.json')
+    return if labels.empty?
+
+    JSON.parse(labels).each do |label|
+      @labels.push(Label.new(label['title'], label['color']))
+    end
```
Comment (Yonashenok) [inline src/modules/book_loader.rb:37]: kindly Check the the `File.write` for `label.json` file work correctly 
- i have noticed that you are saving the color `lable` in `json` file as `colour` and when you read the file `lable["color"]`

File: src/modules/book_saver.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+                        label: { title: book.label.title, color: book.label.color } })
+    end
+    File.write('data/books.json', JSON.pretty_generate(books_data))
+  end
+
+  def save_labels
+    labels_data = []
+    @labels.each do |label|
+      labels_data.push({ title: label.title, colour: label.color })
+    end
```
Comment (Yonashenok) [inline src/modules/book_saver.rb:18]: kindly Check the the `File.write` for `label.json` file work correctly 
- i have noticed that you are saving the color `lable` in `json` file as `colour` and when you read the file `lable["color"]`

Comment (rubydevi) [reply src/modules/book_saver.rb:18]: Thanks for noticing this, I misspelt `color` as `colour`. I will make the change and push again.

**Comment by Yonashenok** [comment]
Hi @rubydevi  ,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966

# To Highlight 💪🏻

- Changes Implemented ✔️ 


 Good job implementing the requirements  :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

**Pull request merged**